server/remote_asset/fetch_server/fetch_server.go

```diff
@@ -126,28 +129,46 @@ func (p *FetchServer) FetchBlob(ctx context.Context, req *rapb.FetchBlobRequest)
 		}
 	}
 	httpClient := timeoutHTTPClient(ctx, req.GetTimeout())
+
+	// Keep track of the last fetch error so that if we fail to fetch, we at
+	// least have something we can return to the client.
+	var fetchErr error
+	setFetchErr := func(err error) {
```

Comment:
This code would be slightly cleaner if you moved the fetching code for a single URL from inside the loop into a separate func.

Reply:
good call! done